x-pack/legacy/plugins/upgrade_assistant/server/np_ready/routes/cluster_checkup.ts

```diff
@@ -4,38 +4,35 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import Boom from 'boom';
 import _ from 'lodash';
-
-import { ServerShim } from '../types';
+import { ServerShimWithRouter } from '../types';
 import { getUpgradeAssistantStatus } from '../lib/es_migration_apis';
-import { EsVersionPrecheck } from '../lib/es_version_precheck';
+import { versionCheckHandlerWrapper } from '../lib/es_version_precheck';
 
 import { createRequestShim } from './create_request_shim';
 
-export function registerClusterCheckupRoutes(server: ServerShim) {
+export function registerClusterCheckupRoutes(server: ServerShimWithRouter) {
   const { callWithRequest } = server.plugins.elasticsearch.getCluster('admin');
   const isCloudEnabled = _.get(server.plugins, 'cloud.config.isCloudEnabled', false);
 
-  server.route({
-    path: '/api/upgrade_assistant/status',
-    method: 'GET',
-    options: {
-      pre: [EsVersionPrecheck],
+  server.router.get(
```

Comment:
Nit: Because this mixes NP and legacy dependencies and the parameter/type is called `server` it's non-obvious from reading the code if you're using legacy, a shimmed facade or NP API's here.

Reply:
I think that was partly the objective here 😅 . Ultimately this code shouldn't have to change much down the line.